File: exercism/python/collatz-conjecture/collatz_conjecture.py
"""
Collatz Conjecture
"""


def steps(number):
    """
    Take any positive integer n. If n is even, divide n by 2
    to get n / 2. If n is odd, multiply n by 3 and add 1 to
    get 3n + 1. Repeat the process indefinitely.

    return the number of steps required to reach 1.
    """

    count = 0

    if number <= 0:
        raise ValueError('Only positive integers are allowed')
    while number != 1:
        count += 1
        # Conditional expression
        number = 3 * number + 1 if number % 2 else number / 2

    return count

File: exercism/go/collatz-conjecture/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(number int) (int, error) {
	if number < 1 {
		return 0, errors.New("values less than 1 not valid Collatz Conjecture input")
	}
	count := 0

	for number != 1 {
		count += 1
		if number%2 != 0 {
			number = 3*number + 1
		} else {
			number /= 2
		}
	}
	return count, nil
}
